packages/ggsmark/__tests__/index.spec.js

```diff
@@ -318,3 +316,98 @@ describe('do not render custom html', () => {
+    `)
+  })
+
+  test('content before and after', () => {
+    // Arrange
+    let string =
+      dedent('**bold text** before text !(https://www.twitch.tv/videos/726762096) after text **bold**',​ ​{​parents​ = ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​})
+
+    // Act
+    let result = ggsmark(string)
+
+    // Assert
+    expect(result).toBe(dedent`
+    <p><strong>bold text</strong> before text <iframe src="https://www.player.twitch.tv/?parent=ggs.sx&#x26;video=726762096" width="620" height="378" allowfullscreen frameborder="0"></iframe> after text <strong>bold</strong></p>
+    `)
+  })
+
+  test('multiple occurrences', () => {
+    // Arrange
+    let string = dedent(`
+    !(https://www.twitch.tv/videos/726762096)
+    !(https://www.twitch.tv/videos/726762096)
+    !(https://www.twitch.tv/videos/726762096)
+    `,​ ​{​parents​: ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​)
+
+    // Act
+    let result = ggsmark(string)
+
+    // Assert
+    expect(result).toBe(dedent`
+    <iframe src="https://www.player.twitch.tv/?parent=ggs.sx&#x26;video=726762096" width="620" height="378" allowfullscreen frameborder="0"></iframe>
+    <iframe src="https://www.player.twitch.tv/?parent=ggs.sx&#x26;video=726762096" width="620" height="378" allowfullscreen frameborder="0"></iframe>
+    <iframe src="https://www.player.twitch.tv/?parent=ggs.sx&#x26;video=726762096" width="620" height="378" allowfullscreen frameborder="0"></iframe>
+    `)
+  })
+})
+
+describe('render twitch clip blocks', () => {
+  test('single line', () => {
+    // Arrange
+    let string = dedent(`!(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)`,​ ​{​parents​ = ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​)
+
+    // Act
+    let result = ggsmark(string)
+
+    // Assert
+    expect(result).toBe(dedent`
+    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;clip=/LovelyAstuteCoffeeImGlitch" width="620" height="378" allowfullscreen frameborder="0"></iframe>
+    `)
+  })
+
+  test('content before and after', () => {
+    // Arrange
+    let string =
+      dedent('**bold text** before text !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch) after text **bold**',​ ​{​parents​ = ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​})
+    // Act
+    let result = ggsmark(string)
+
+    // Assert
+    expect(result).toBe(dedent`
+    <p><strong>bold text</strong> before text <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;clip=/LovelyAstuteCoffeeImGlitch" width="620" height="378" allowfullscreen frameborder="0"></iframe> after text <strong>bold</strong></p>
+    `)
+  })
+
+  test('multiple occurrences', () => {
+    // Arrange
+    let string = dedent(`
+    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
+    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
+    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
+    `,​ ​{​parents​ = ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​})
+
+    // Act
+    let result = ggsmark(string)
```

Comment:
```suggestion
    let result = ggsmark(string,​ ​{​parents​ = ​[​'ggs.sx'​,​ ​'example.com'​,​ ​'blah.blah'​]​})
```
Options go here, not the string itself. Imagine using this package and configuring it to your liking